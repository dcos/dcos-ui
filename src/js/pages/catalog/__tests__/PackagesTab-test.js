/* eslint-disable no-unused-vars */
import React from "react";
/* eslint-enable no-unused-vars */
import { mount } from "enzyme";

jest.mock("../../../utils/ScrollbarUtil");
jest.mock("../../../components/Page", function() {
  const Page = ({ children }) => <div>{children}</div>;
  Page.Header = ({ children }) => <div>{children}</div>;

  return Page;
});

// Setting useFixtures for when we load CosmosPackagesStore/CosmosPackageActions
/* eslint-disable import/newline-after-import */
const Config = require("../../../config/Config");
var configUseFixtures = Config.useFixtures;
Config.useFixtures = true;
const CosmosPackagesStore = require("../../../stores/CosmosPackagesStore");
Config.useFixtures = configUseFixtures;
/* eslint-enable import/newline-after-import */

const PackagesTab = require("../PackagesTab");
const UniversePackagesList = require("../../../structs/UniversePackagesList");

const renderer = require("react-test-renderer");

let thisInstance,
  thisCosmosPackagesStoreGetAvailablePackages,
  thisPackages,
  thisGetAvailablePackages,
  thisFetchAvailablePackages;
describe("PackagesTab", function() {
  beforeEach(function() {
    thisInstance = mount(<PackagesTab />);
  });

  describe("#handleDetailOpen", function() {
    beforeEach(function() {
      thisInstance.instance().handleDetailOpen = jasmine.createSpy(
        "handleDetailOpen"
      );
      jest.runAllTimers();
    });

    it("calls handler when panel is clicked", function() {
      thisInstance.setState({ loading: false });
      thisInstance
        .find(".panel.clickable")
        .at(0)
        .simulate("click");

      expect(
        thisInstance
          .instance()
          .handleDetailOpen.calls.mostRecent()
          .args[0].get("name")
      ).toEqual("arangodb3");
    });
  });

  describe("#getPackageGrid", function() {
    beforeEach(function() {
      thisCosmosPackagesStoreGetAvailablePackages =
        CosmosPackagesStore.getAvailablePackages;
      thisPackages = CosmosPackagesStore.getAvailablePackages();
    });

    afterEach(function() {
      CosmosPackagesStore.getAvailablePackages = thisCosmosPackagesStoreGetAvailablePackages;
    });

    it("returns packages", function() {
      expect(
        thisInstance.instance().getPackageGrid(thisPackages).length
      ).toEqual(97);
    });

    it("doesn't return packages", function() {
      CosmosPackagesStore.getAvailablePackages = function() {
        return new UniversePackagesList();
      };

      const packages = CosmosPackagesStore.getAvailablePackages();
      expect(thisInstance.instance().getPackageGrid(packages).length).toEqual(
        0
      );
    });
  });

  describe("with empty state", function() {
    beforeEach(function() {
      thisGetAvailablePackages = CosmosPackagesStore.getAvailablePackages;
      thisFetchAvailablePackages = CosmosPackagesStore.fetchAvailablePackages;
      CosmosPackagesStore.getAvailablePackages = () => {
        return new UniversePackagesList();
      };
      CosmosPackagesStore.fetchAvailablePackages = () => {};
    });

    afterEach(function() {
      CosmosPackagesStore.getAvailablePackages = thisGetAvailablePackages;
      CosmosPackagesStore.fetchAvailablePackages = thisFetchAvailablePackages;
    });

    it("displays AlertPanel with action to Package Repositories", function() {
      thisInstance = renderer.create(<PackagesTab />);
      thisInstance.getInstance().onCosmosPackagesStoreAvailableSuccess();

      var tree = thisInstance.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe("with packages on the list", function() {
    it("displays the catalog with packages", function() {
      thisInstance = renderer.create(<PackagesTab />);
      thisInstance.getInstance().onCosmosPackagesStoreAvailableSuccess();

      var tree = thisInstance.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
