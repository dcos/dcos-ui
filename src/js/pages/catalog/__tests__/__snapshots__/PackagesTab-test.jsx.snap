// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PackagesTab with empty state displays AlertPanel with action to Package Repositories 1`] = `
<div>
  <div />
  <div
    className="panel alert-panel text-align-center flush-bottom"
    onClick={undefined}
  >
    <div
      className="panel-content panel-cell panel-cell-borderless panel-cell-wider panel-cell-taller"
    >
      <h2
        className="flush-top"
      >
        <span
          className={undefined}
        >
          No package repositories
        </span>
      </h2>
      <p
        className="tall"
      >
        You need at least one package repository with some packages to be able to install packages. For more 
        <a
          href="https://docs.mesosphere.com/latest/administering-clusters/repo"
          target="_blank"
        >
          information on repositories
        </a>
        .
      </p>
      <div
        className="button-collection flush-bottom"
      >
        <a
          className="button button-primary"
          onClick={[Function]}
          style={Object {}}
        >
          <span
            className={undefined}
          >
            Add Package Repository
          </span>
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`PackagesTab with packages on the list displays the catalog with packages 1`] = `
<div>
  <div />
  <div
    className="container"
  >
    <div
      className="pod flush-horizontal flush-top"
    >
      <div
        className="filter-bar"
      >
        <div
          className="filter-bar-left"
        >
          <div
            className="filter-bar-item"
          >
            <div
              className="form-group flush-bottom"
            >
              <div
                className="form-control form-control-group filter-input-text-group"
                onBlur={[Function]}
                onClick={[Function]}
              >
                <span
                  className="form-control-group-add-on form-control-group-add-on-prepend "
                >
                  <svg
                    aria-label="system-search icon"
                    className="css-1vunjz0"
                    data-cy="icon"
                    height={16}
                    preserveAspectRatio="xMinYMin meet"
                    role="img"
                    viewBox="0 0 16 16"
                    width={16}
                  >
                    <use
                      xlinkHref="#system-search"
                    />
                  </svg>
                </span>
                <input
                  className="form-control filter-input-text"
                  onChange={[Function]}
                  placeholder="Search catalog"
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
          <div
            className="filter-bar-item"
          >
            <span
              className="button-group flush-bottom"
            >
              <button
                className="button button-outline"
                onClick={[Function]}
              >
                <b
                  className={undefined}
                >
                  All
                </b>
                <span
                  className="css-17jd12v"
                  data-cy="badge"
                >
                  <b>
                    97
                  </b>
                </span>
              </button>
              <button
                className="button button-outline active"
                onClick={[Function]}
              >
                <b
                  className={undefined}
                >
                  Certified
                </b>
                <span
                  className="css-17jd12v"
                  data-cy="badge"
                >
                  <b>
                    9
                  </b>
                </span>
              </button>
              <button
                className="button button-outline"
                onClick={[Function]}
              >
                <b
                  className={undefined}
                >
                  Community
                </b>
                <span
                  className="css-17jd12v"
                  data-cy="badge"
                >
                  <b>
                    88
                  </b>
                </span>
              </button>
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      className="pod flush-top flush-horizontal clearfix"
    >
      <h1
        className="short flush-top"
      >
        Certified
      </h1>
      <p
        className="tall flush-top"
      >
        Certified packages are verified by D2iQ for interoperability with DC/OS.
      </p>
      <div
        className="panel-grid row"
      >
        <div
          className="panel-grid-item column-6 column-small-4 column-large-3"
        >
          <div
            className="panel panel-interactive clickable"
            onClick={[Function]}
          >
            <div
              className="panel-header panel-cell panel-cell-narrow panel-cell-borderless horizontal-center flush-bottom"
            >
              <div
                className="icon icon-jumbo icon-image-container icon-app-container icon-app-container--borderless icon-default-white"
              >
                <img
                  className=""
                  onError={[Function]}
                  src={Object {}}
                />
              </div>
            </div>
            <div
              className="panel-content panel-cell panel-cell-narrow panel-cell-short panel-cell-borderless horizontal-center text-align-center"
            >
              <div
                className="h3 flush"
              >
                arangodb3
              </div>
              <small
                className="flush"
              >
                1.4.6
              </small>
            </div>
            <div
              className="panel-footer panel-cell panel-cell-narrow flush-top horizontal-center"
            >
              <span
                className="css-11hyrb9"
                data-cy="badge"
              >
                Certified
              </span>
            </div>
          </div>
        </div>
        <div
          className="panel-grid-item column-6 column-small-4 column-large-3"
        >
          <div
            className="panel panel-interactive clickable"
            onClick={[Function]}
          >
            <div
              className="panel-header panel-cell panel-cell-narrow panel-cell-borderless horizontal-center flush-bottom"
            >
              <div
                className="icon icon-jumbo icon-image-container icon-app-container icon-app-container--borderless icon-default-white"
              >
                <img
                  className=""
                  onError={[Function]}
                  src={Object {}}
                />
              </div>
            </div>
            <div
              className="panel-content panel-cell panel-cell-narrow panel-cell-short panel-cell-borderless horizontal-center text-align-center"
            >
              <div
                className="h3 flush"
              >
                artifactory
              </div>
              <small
                className="flush"
              >
                5.1.4
              </small>
            </div>
            <div
              className="panel-footer panel-cell panel-cell-narrow flush-top horizontal-center"
            >
              <span
                className="css-11hyrb9"
                data-cy="badge"
              >
                Certified
              </span>
            </div>
          </div>
        </div>
        <div
          className="panel-grid-item column-6 column-small-4 column-large-3"
        >
          <div
            className="panel panel-interactive clickable"
            onClick={[Function]}
          >
            <div
              className="panel-header panel-cell panel-cell-narrow panel-cell-borderless horizontal-center flush-bottom"
            >
              <div
                className="icon icon-jumbo icon-image-container icon-app-container icon-app-container--borderless icon-default-white"
              >
                <img
                  className=""
                  onError={[Function]}
                  src={Object {}}
                />
              </div>
            </div>
            <div
              className="panel-content panel-cell panel-cell-narrow panel-cell-short panel-cell-borderless horizontal-center text-align-center"
            >
              <div
                className="h3 flush"
              >
                cassandra
              </div>
              <small
                className="flush"
              >
                1.0.25-3.0.10
              </small>
            </div>
            <div
              className="panel-footer panel-cell panel-cell-narrow flush-top horizontal-center"
            >
              <span
                className="css-11hyrb9"
                data-cy="badge"
              >
                Certified
              </span>
            </div>
          </div>
        </div>
        <div
          className="panel-grid-item column-6 column-small-4 column-large-3"
        >
          <div
            className="panel panel-interactive clickable"
            onClick={[Function]}
          >
            <div
              className="panel-header panel-cell panel-cell-narrow panel-cell-borderless horizontal-center flush-bottom"
            >
              <div
                className="icon icon-jumbo icon-image-container icon-app-container icon-app-container--borderless icon-default-white"
              >
                <img
                  className=""
                  onError={[Function]}
                  src={Object {}}
                />
              </div>
            </div>
            <div
              className="panel-content panel-cell panel-cell-narrow panel-cell-short panel-cell-borderless horizontal-center text-align-center"
            >
              <div
                className="h3 flush"
              >
                confluent-kafka
              </div>
              <small
                className="flush"
              >
                1.1.19-3.2.0e
              </small>
            </div>
            <div
              className="panel-footer panel-cell panel-cell-narrow flush-top horizontal-center"
            >
              <span
                className="css-11hyrb9"
                data-cy="badge"
              >
                Certified
              </span>
            </div>
          </div>
        </div>
        <div
          className="panel-grid-item column-6 column-small-4 column-large-3"
        >
          <div
            className="panel panel-interactive clickable"
            onClick={[Function]}
          >
            <div
              className="panel-header panel-cell panel-cell-narrow panel-cell-borderless horizontal-center flush-bottom"
            >
              <div
                className="icon icon-jumbo icon-image-container icon-app-container icon-app-container--borderless icon-default-white"
              >
                <img
                  className=""
                  onError={[Function]}
                  src={Object {}}
                />
              </div>
            </div>
            <div
              className="panel-content panel-cell panel-cell-narrow panel-cell-short panel-cell-borderless horizontal-center text-align-center"
            >
              <div
                className="h3 flush"
              >
                elastic
              </div>
              <small
                className="flush"
              >
                1.0.8-5.2.2
              </small>
            </div>
            <div
              className="panel-footer panel-cell panel-cell-narrow flush-top horizontal-center"
            >
              <span
                className="css-11hyrb9"
                data-cy="badge"
              >
                Certified
              </span>
            </div>
          </div>
        </div>
        <div
          className="panel-grid-item column-6 column-small-4 column-large-3"
        >
          <div
            className="panel panel-interactive clickable"
            onClick={[Function]}
          >
            <div
              className="panel-header panel-cell panel-cell-narrow panel-cell-borderless horizontal-center flush-bottom"
            >
              <div
                className="icon icon-jumbo icon-image-container icon-app-container icon-app-container--borderless icon-default-white"
              >
                <img
                  className=""
                  onError={[Function]}
                  src={Object {}}
                />
              </div>
            </div>
            <div
              className="panel-content panel-cell panel-cell-narrow panel-cell-short panel-cell-borderless horizontal-center text-align-center"
            >
              <div
                className="h3 flush"
              >
                gitlab
              </div>
              <small
                className="flush"
              >
                1.0-9.0.0
              </small>
            </div>
            <div
              className="panel-footer panel-cell panel-cell-narrow flush-top horizontal-center"
            >
              <span
                className="css-11hyrb9"
                data-cy="badge"
              >
                Certified
              </span>
            </div>
          </div>
        </div>
        <div
          className="panel-grid-item column-6 column-small-4 column-large-3"
        >
          <div
            className="panel panel-interactive clickable"
            onClick={[Function]}
          >
            <div
              className="panel-header panel-cell panel-cell-narrow panel-cell-borderless horizontal-center flush-bottom"
            >
              <div
                className="icon icon-jumbo icon-image-container icon-app-container icon-app-container--borderless icon-default-white"
              >
                <img
                  className=""
                  onError={[Function]}
                  src={Object {}}
                />
              </div>
            </div>
            <div
              className="panel-content panel-cell panel-cell-narrow panel-cell-short panel-cell-borderless horizontal-center text-align-center"
            >
              <div
                className="h3 flush"
              >
                jenkins
              </div>
              <small
                className="flush"
              >
                3.0.3-2.32.3
              </small>
            </div>
            <div
              className="panel-footer panel-cell panel-cell-narrow flush-top horizontal-center"
            >
              <span
                className="css-11hyrb9"
                data-cy="badge"
              >
                Certified
              </span>
            </div>
          </div>
        </div>
        <div
          className="panel-grid-item column-6 column-small-4 column-large-3"
        >
          <div
            className="panel panel-interactive clickable"
            onClick={[Function]}
          >
            <div
              className="panel-header panel-cell panel-cell-narrow panel-cell-borderless horizontal-center flush-bottom"
            >
              <div
                className="icon icon-jumbo icon-image-container icon-app-container icon-app-container--borderless icon-default-white"
              >
                <img
                  className=""
                  onError={[Function]}
                  src={Object {}}
                />
              </div>
            </div>
            <div
              className="panel-content panel-cell panel-cell-narrow panel-cell-short panel-cell-borderless horizontal-center text-align-center"
            >
              <div
                className="h3 flush"
              >
                marathon
              </div>
              <small
                className="flush"
              >
                1.4.2
              </small>
            </div>
            <div
              className="panel-footer panel-cell panel-cell-narrow flush-top horizontal-center"
            >
              <span
                className="css-11hyrb9"
                data-cy="badge"
              >
                Certified
              </span>
            </div>
          </div>
        </div>
        <div
          className="panel-grid-item column-6 column-small-4 column-large-3"
        >
          <div
            className="panel panel-interactive clickable"
            onClick={[Function]}
          >
            <div
              className="panel-header panel-cell panel-cell-narrow panel-cell-borderless horizontal-center flush-bottom"
            >
              <div
                className="icon icon-jumbo icon-image-container icon-app-container icon-app-container--borderless icon-default-white"
              >
                <img
                  className=""
                  onError={[Function]}
                  src={Object {}}
                />
              </div>
            </div>
            <div
              className="panel-content panel-cell panel-cell-narrow panel-cell-short panel-cell-borderless horizontal-center text-align-center"
            >
              <div
                className="h3 flush"
              >
                spark
              </div>
              <small
                className="flush"
              >
                1.0.9-2.1.0-1
              </small>
            </div>
            <div
              className="panel-footer panel-cell panel-cell-narrow flush-top horizontal-center"
            >
              <span
                className="css-11hyrb9"
                data-cy="badge"
              >
                Certified
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
